lib/memory_mosq.h

```diff
@@ -20,7 +20,7 @@ and the Eclipse Distribution License is available at
 #include <stdio.h>
 #include <sys/types.h>
 
-#if defined(WITH_MEMORY_TRACKING) && defined(WITH_BROKER) && !defined(WIN32) && !defined(__SYMBIAN32__) && !defined(__ANDROID__) && !defined(__UCLIBC__) && !defined(__OpenBSD__)
+#if defined(WITH_MEMORY_TRACKING) && defined(WITH_BROKER) && !defined(WIN32) && !defined(__SYMBIAN32__) && !defined(__ANDROID__) && !defined(__UCLIBC__) && !defined(__OpenBSD__) && !defined(__NetBSD__)
```

Comment:
so freebsd supports memory tracking, but not openbsd or netbsd.  Perhaps this is really a check on libc versions?  (much like the existing uclibc check?)